core/jvm/src/main/scala/fs2/compression.scala

```diff
@@ -1222,10 +1254,17 @@ object compression {
     1024 * 1024 // A limit is good practice. Actual limit will be max(chunk.size, soft limit). 1 MiB feels reasonable for a comment.
 
   private def asChunkBytes(values: Array[Byte]): Chunk[Byte] =
-    asChunkBytes(values, values.length)
-
-  private def asChunkBytes(values: Array[Byte], length: Int): Chunk[Byte] =
-    if (length > 0) Chunk.Bytes(values, 0, length) else Chunk.empty[Byte]
+    asChunkBytes(values, values.length, false)
```

Comment:
Is this always safe? Given the buffer sharing, it's not clear to me at first glance. Safe because only call site is from header processing, which doesn't share a buffer?

Reply:
Yes, that was the reasoning. I guess could be split, renamed to be more dangerous sounding.., and in that place it would be obvious .. Or prefixed move vs copy